src/main/java/com/zzpj/dtos/AddBookDto.java

```diff
@@ -0,0 +1,22 @@
+package com.zzpj.dtos;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotEmpty;
+
+@NoArgsConstructor
+@Data
+@JsonPropertyOrder({"name", "author", "category"})
+public class AddBookDto {
+
+    @NotEmpty(message = "name cannot be null!")
+    public String name;
+
+    @NotEmpty(message = "author cannot be null!")
+    public String author;
+
+    public Long category;
+
```

Comment:
Nie wiem czy jest sens brnąć w te stronę, bo zaczniemy tonąć w tych Dto, bo nie długo może zajść potrzeba dodanie n kolejnych dla kilku różnych modeli :P Myślę, że można na spokojnie zrobić pola tylko z CategoryDto, a Dtosy będą miały Id, a pola można walidować na poziomie usług. Znaczy to moja propozycja :P

Reply:
Drugie rozwiązanie bardziej mi się podoba :P 
Po prostu teraz nam się tak wydaje, ale z tym pierwszym rozwiązaniem jest naprawdę łatwo popłynąć :P